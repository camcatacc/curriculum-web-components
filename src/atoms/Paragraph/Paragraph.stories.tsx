// Modules
import React from "react";

// Storybook Modules
import { ComponentStory, ComponentMeta } from "@storybook/react";

// Elements
import Paragraph from "atoms/Paragraph/Paragraph";

// Template
const Template: ComponentStory<typeof Paragraph> = (args) => <Paragraph {...args} />;

// Variants
export const Body = Template.bind({});
Body.args = {
	children: "This is a body1 paragraph",
	variant: "body1"
};

export const Body2 = Template.bind({});
Body2.args = {
	children: "This is a body2 paragraph",
	variant: "body2"
};

// Export
export default {
	title: "Atoms/Paragraph",
	component: Paragraph
} as ComponentMeta<typeof Paragraph>;
