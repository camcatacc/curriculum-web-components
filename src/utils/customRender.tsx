import { ThemeProvider } from "@material-ui/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { ReactElement } from "react";
import { getMaterialTheme } from "styles/materialUi";

export const customRender = (ui: ReactElement, renderOptions?: RenderOptions & { darkMode?: boolean }): RenderResult => {
	const Wrapper: React.FC = ({ children }) => {
		return <ThemeProvider theme={getMaterialTheme(renderOptions?.darkMode ? "dark" : "light")}>{children}</ThemeProvider>;
	};

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};
