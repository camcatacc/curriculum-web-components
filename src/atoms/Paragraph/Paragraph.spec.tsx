// Modules
import React from "react";
import { customRender } from "utils/customRender";

// Accessibility
import { axe, toHaveNoViolations } from "jest-axe";

// Elements
import Paragraph from "atoms/Paragraph/Paragraph";

// Definitions
import type { ParagraphProps } from "atoms/Paragraph/Paragraph";

// Default props
const defaultProps: ParagraphProps = {
	children: <p>Bu</p>
};

// Tests
expect.extend(toHaveNoViolations);
describe("Paragraph", () => {
	it("Renders as expected and has no accessibility violations", async () => {
		const { container } = customRender(<Paragraph {...defaultProps} />);
		expect(container).toMatchSnapshot();
		expect(await axe(container)).toHaveNoViolations();
	});
});
