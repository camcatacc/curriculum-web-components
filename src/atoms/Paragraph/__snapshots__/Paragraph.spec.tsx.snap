// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Renders as expected and has no accessibility violations 1`] = `
<div>
  <p
    class="MuiTypography-root sc-bdvvtL gedMyf MuiTypography-body1 MuiTypography-alignJustify"
  >
    Bu
  </p>
</div>
`;
